dataprofiler/data_readers/data_utils.py

```diff
@@ -610,3 +612,54 @@ def is_stream_buffer(filepath_or_buffer):
     if isinstance(filepath_or_buffer, (StringIO, BytesIO)):
         return True
     return False
+
+def is_valid_url(url_as_string):
+    """
+    Determines whether a given string is a valid URL
+
+    :param url_as_string: string to be tested if URL 
+    :type url_as_string: str
+    :return: true if string is a valid URL
+    :rtype: boolean
+    """
+    result = urllib.parse.urlparse(url_as_string)
+    # this is the minimum characteristics needed for a valid URL
+    return all([result.scheme, result.netloc])
+
+
+def url_to_bytes(url_as_string, options):
+    """
+    Reads in URL and converts it to a byte stream
+
+    :param url_as_string: string to read as URL 
+    :type url_as_string: str
+    :return: true if string is a valid URL
+    :rtype: boolean
+    """
+    stream = BytesIO()
+
+    verify_url = False
+    if 'verify' in options:
+        verify_url = options['verify']
+
+    with requests.get(url_as_string, stream=True, verify=verify_url) as url:
+        url.raise_for_status()
+        if 'Content-length' in url.headers and \
+            int(url.headers['Content-length']) >= 1024 ** 3:
+
+            raise ValueError('The downloaded file from the url may not be \
+                larger than 1GB')
+
+        total_bytes = 0
+        c_size = 8192
+
+        for chunk in url.iter_content(chunk_size=8192):
```

Comment:
Should probably pass `c_size` directly here in case the value ever gets changed.